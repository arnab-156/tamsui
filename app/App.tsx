import React from 'react';

import AppRoutes from './AppRoutes';

export default function App(): React.ReactElement {
  return (
    <React.StrictMode>
      <html lang="en">
        <head>
          <title>Hello, React SSR!</title>
        </head>
        <body>
          <AppRoutes />
        </body>
      </html>
    </React.StrictMode>
  );
}
