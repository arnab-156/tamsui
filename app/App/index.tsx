import React from 'react';
import 'app/global.module.scss';

type AppProps = React.PropsWithChildren;

export default function App({ children }: AppProps): React.ReactElement {
  return (
    <React.StrictMode>
      <html lang="en">
        <head>
          <title>Hello, React SSR!</title>
        </head>
        <body>
          { children }
        </body>
      </html>
    </React.StrictMode>
  );
}
