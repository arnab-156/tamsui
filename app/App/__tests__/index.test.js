import React from 'react';
import renderer from 'react-test-renderer';

import App from '../index';

jest.mock('app/Routes', function mockAppRoutes() {
  return function mockedAppRoutes() {
    return (
      <main>
        <h1>Totally Kickass Website</h1>
      </main>
    );
  };
});

describe('App Component', () => {
  test('matches snapshot', () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
