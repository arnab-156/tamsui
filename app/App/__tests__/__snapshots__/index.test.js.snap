// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches child snapshot 1`] = `
<html
  lang="en"
>
  <head>
    <title>
      Hello, React SSR!
    </title>
    <link
      href="/Root CSS module"
      rel="stylesheet"
      type="text/css"
    />
  </head>
  <body>
    <h1>
      App Child Component
    </h1>
  </body>
</html>
`;

exports[`App Component matches no-child snapshot 1`] = `
<html
  lang="en"
>
  <head>
    <title>
      Hello, React SSR!
    </title>
    <link
      href="/Root CSS module"
      rel="stylesheet"
      type="text/css"
    />
  </head>
  <body />
</html>
`;
